src/ui/components/Shared/TracksList.react.js

```diff
@@ -40,25 +43,27 @@ export default class TracksList extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      selected  : [],
-      scrollTop : 0,
+      selected: [],
+      tilesScrolled: 0,
     };
 
-    this.showContextMenu  = this.showContextMenu.bind(this);
-    this.startPlayback    = this.startPlayback.bind(this);
-    this.scrollTracksList = this.scrollTracksList.bind(this);
-    this.selectTrack      = this.selectTrack.bind(this);
-    this.onKey            = this.onKey.bind(this);
-
-    this.rowHeight = 30;
+    this.showContextMenu = this.showContextMenu.bind(this);
+    this.startPlayback = this.startPlayback.bind(this);
+    this.onScroll = this.onScroll.bind(this);
+    this.selectTrack = this.selectTrack.bind(this);
+    this.onKey = this.onKey.bind(this);
   }
 
   componentDidMount() {
     this.renderView = document.querySelector('.tracks-list-render-view');
   }
 
-  scrollTracksList() {
-    this.setState({ scrollTop : this.renderView.scrollTop });
+  onScroll() {
+    const tilesScrolled = Math.floor(this.renderView.scrollTop / TILE_HEIGHT);
+
+    if (this.state.tilesScrolled !== tilesScrolled) {
```

Comment:
Are you sure this is not checked by React itself?
Like, does React re-render components if you set state with exactly the same values?

Reply:
Unfortunately yes, I just used this to avoid having to implement a complex `shouldComponentUpdate` https://stackoverflow.com/questions/24718709/reactjs-does-render-get-called-any-time-setstate-is-called#24719289

Reply:
Weird, [React docs say that setState is async inside event handlers](https://reactjs.org/docs/faq-state.html#when-is-setstate-asynchronous). 
`onScroll` seems like event handler to me

Reply:
`setState` is async indeed, but if you `this.setState({a: 5})` and `this.state.a` already equals 5, `render()` will be re-triggered.